metron-platform/metron-elasticsearch/src/main/java/org/apache/metron/elasticsearch/dao/ElasticsearchRetrieveLatestDao.java

```diff
@@ -102,54 +92,47 @@ public Document getLatest(String guid, String sensorType) throws IOException {
     if (guids == null || guids.isEmpty()) {
       return Collections.emptyList();
     }
-    QueryBuilder query = null;
-    IdsQueryBuilder idsQuery;
-    if (sensorTypes != null) {
-      String[] types = sensorTypes.stream().map(sensorType -> sensorType + "_doc")
-          .toArray(String[]::new);
-      idsQuery = QueryBuilders.idsQuery(types);
-    } else {
-      idsQuery = QueryBuilders.idsQuery();
-    }
 
-    for (String guid : guids) {
-      query = idsQuery.addIds(guid);
+    // should match any of the guids
+    // the 'guid' field must be of type 'keyword' or this term query will not match
+    BoolQueryBuilder guidQuery = boolQuery().must(termsQuery(Constants.GUID, guids));
```

Comment:
We can no longer use an id query, since the doc ID is no longer the same as the GUID.